src/components/ConnectionSidebar.tsx

```diff
@@ -49,12 +59,40 @@ const ConnectionSidebar = () => {
         .getOrFetchDatabaseList(currentConnectionCtx.connection)
         .finally(() => {
           setIsRequestingDatabase(false);
+          const database = databaseList.find(
+            (database) =>
+              database.name ===
+              useConnectionStore.getState().currentConnectionCtx?.database?.name
+          );
+          if (database) {
+            tableSchemaLoadingState.setLoading();
+            connectionStore.getOrFetchDatabaseSchema(database).then(() => {
+              tableSchemaLoadingState.setFinish();
+            });
+          }
         });
     } else {
       setIsRequestingDatabase(false);
     }
   }, [currentConnectionCtx?.connection]);
 
+  useEffect(() => {
+    const tableList =
+      connectionStore.databaseList.find(
+        (database) =>
+          database.connectionId === currentConnectionCtx?.connection.id &&
+          database.name === currentConnectionCtx?.database?.name
+      )?.tableList || [];
+
+    updateTableList([
+      {
+        name: t("connection.all-tables"),
```

Comment:
To make it less error prone.

this should be name: ""

and in the the rendering part if name == "" then display t("connection.all-tables")